index.js

```diff
@@ -797,17 +797,16 @@
   //. the type comprising every [`String`][] value except `''`.
   //.
   //. The given type must satisfy the [Monoid][] and [Setoid][] specifications.
-  var NonEmpty = UnaryType(
-    'sanctuary-def/NonEmpty',
-    functionUrl('NonEmpty'),
-    Any,
-    function(x) {
+  function NonEmpty(parent) {
+    function test(x) {
       return Z.Monoid.test(x) &&
              Z.Setoid.test(x) &&
              !Z.equals(x, Z.empty(x.constructor));
-    },
-    function(monoid) { return [monoid]; }
-  );
+    }
+    function extract(monoid) { return [monoid]; }
+    var t = UnaryTypeWithUrl('sanctuary-def/NonEmpty', parent, test, extract);
```

Comment:
The problem is that

```javascript
$.NonEmpty($.Array($.String)).validate([1, 2, 3]).value.propPath
```

evaluates to `['$1']` rather than `['$1', '$1']` because we first validate the parent type and, if validation fails, make no adjustment to `propPath` to account for the extra level of nesting introduced by `$.NonEmpty`.